### PR TITLE
change output binary files

### PR DESCRIPTION
**Pull request by ewoodev**
# commit 1
    tools/set_bininfo: fix set kernel binary name when without board_metadata.txt
    
    It is setting kernel binary name to "tinyara.bin" when without board_metadata.txt file.
    But save_bininfo function is not saving kernel name when without borad_betadata.txt file.
    
    So This commit adds case of when bin_name is nothing.

# commit 2
    Makefile.unix, os/tools, build/configs: change output binary files
    This commit has 3 goal as below
    1. get a binary with samsung header of board specific binary file name.
    2. get a binary without samsung header for debugging.
    3. set_bininfo.py converts trpk file using only finally binary file(finished include samsung header).
    
    More detail about binary file name of each step as below.
    1. call MAKE_BOARD_SPECIFIC_BIN
	    - make board specific binary file
	    - file name : Each board has a different file name (ex, rtl8721csm : km0_km4_image2.bin)
    2. call MAKE_SAMSUNG_HEADER
	    a. copy and rename original board specific binary file to {filename}_without_header.bin
		    (ex, rtl8721csm : km0_km4_image2.bin -> km0_km4_image2_without_header.bin)
	    b. include samsung header at board specific binary file(keep the filename of board specific name)
		    (ex, rtl8721csm : km0_km4_image2.bin -> km0_km4_image2.bin)
    3. set_bininfo.py
	    - copy and rename binary file including samsung header to kernel_{board name}_{YYMMDD}.trpk
		    (ex, rtl8721csm : km0_km4_image2.bin -> kernel_rtl8721csm_220109.trpk)

# commit 3
    dutils.sh, Makefile.unix: change app name to getting from .config
    the app binary is made into filename of CONFIG_APP1_BIN_NAME in Makefile of loadable_apps directory
    but app binary filename which is a hard copied file name as 'app1','app2' is used in post step of Makfile.unix and dutil.sh
    
    Therefore, This commit changes app name to getting from .config

# commit 4
    os: Separate setting .bininfo and converting binary
    This commit separates setting binary name and converting binary in set_bininfo.py
    
    The set_bininfo.py has the role of saving the binary name in a .bininfo file.
    The convert_binary.py is created to convert the binary.

# commit 5
    os: Fix to include Samsung header in hex, srec binary
    The $BIN.bin file is used to convert to hex or srec binary.
    But if binary name is changed in MAKE_BOARD_SPECIFIC_BIN, This use binary that
    is not board specific binary, also the using binary is without samsung header.
    
    So This commit changes to use binary with samsung header when convert bin to hex or srec.

### CODE REVIEW

File: os/tools/mkbinheader.py [hunk 1]
```diff
@@ -350,6 +351,11 @@ def make_common_binary_header():
 ############################################################################
 file_path  = sys.argv[1]
 binary_type = sys.argv[2]
+
+# copy orignal binary without hearder
```
Comment (abhishek-samsung) [inline os/tools/mkbinheader.py:355]: typo hearder -> header

Comment (ewoodev) [reply os/tools/mkbinheader.py]: thanks~ done

File: os/tools/set_bininfo.py [hunk 2]
```diff
@@ -37,12 +37,14 @@ def save_bininfo(bin_name) :
...
+        elif ("app1" in bin_name) :
             f.write('APP1_BIN_NAME=' + bin_name + '\n')
-        if ("app2" in bin_name) :
+        elif ("app2" in bin_name) :
             f.write('APP2_BIN_NAME=' + bin_name + '\n')
-        if ("common" in bin_name) :
+        elif ("common" in bin_name) :
             f.write('COMMON_BIN_NAME=' + bin_name + '\n')
+        else :
+            f.write('KERNEL_BIN_NAME=' + bin_name + '\n')
```
[the remark this answers is not in the document]
Comment (ewoodev) [reply os/tools/set_bininfo.py:44]: yes, I changed!

File: docs/HowToPortTizenRTBinaryHeader.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 # How To Port TizenRT Binary Header
 
+
```
[the remark this answers is not in the document]
Comment (ewoodev) [reply docs/HowToPortTizenRTBinaryHeader.md]: Thanks Done!

File: docs/HowToPortTizenRTBinaryHeader.md [hunk 4]
```diff
@@ -51,25 +52,36 @@ The binary header will be added at front of TizenRT binary and it will be locate
 
 ## How to add Binary Header
```
[the remark this answers is not in the document]
Comment (ewoodev) [reply docs/HowToPortTizenRTBinaryHeader.md:52]: Thanks. I changed explains

File: os/dutils.sh [hunk 5]
```diff
@@ -78,14 +78,21 @@ function GET_SPECIFIC_DOCKER_IMAGE()
...
+	source $CONFIGFILE
+	if [ ! -z $CONFIG_APP1_BIN_NAME ]; then
+		APP1_DBG_BIN="${CONFIG_APP1_BIN_NAME}_dbg"
+	fi
+	if [ ! -z $CONFIG_APP2_BIN_NAME ]; then
+		APP2_DBG_BIN="${CONFIG_APP2_BIN_NAME}_dbg"
+	fi
+	if [ ! -z $CONFIG_COMMON_BINARY_NAME ]; then
+		CMN_DBG_BIN="${CONFIG_COMMON_BINARY_NAME}_dbg"
+	fi
```
[the remark this answers is not in the document]
Comment (ewoodev) [reply os/dutils.sh:95]: The reason is for matching dutils.sh script.
If CONFIG_APP1_BIN_NAME is disabled, The script dutil.sh print `_dbg`. 

If you select `2.APP1` -> `1.nm` in dutils.sh, `"Binary file not found: app1_dbg"`  is printed

but if don't check CONFIG_APP1_BIN_NAME, `"Binary file not found: _dbg"`  is printed

**Pull request merged**